### DISCUSSION TITLE
Workarounds for dynamically generating datasets (assets)

### DISCUSSION THREAD

**Discussion by dmsfabiano** [dagster-io/dagster#15946] [Q&A] [answered]
More of a question in regards to dagster concepts. In the GitHub discussions in mentions that Dynamic Assets are not supported. 

I have a simple use case where I am trying to figure out if I can accomplish this with dagster. Basically I was thinking of defining our ML Model training truth tables as Assets, those assets could be used for ML experiments (dagster jobs) with specific truth tables (assets).I have a "datasets" table on PostgreSQL (ConfigurableResource), for which I was planning to query and create an "Datasets" assets which is basically a list of datasets names.

Originally what I wanted is to dynamically create an asset out of those. We get a number of truth tables / datasets each month so Ideally I wouldn't want to push code changes just to add 1 asset each time we get another dataset

Is there another way to change the problem statement to use the currently available tools? I really like the concept of software defined data assets

There is also the idea of using Dynamic Partitions, but I would need to independently reference partitions as dependencies for other jobs or assets (i.e. Partition.2 is a dependency of Job A, but Partition.3 is a dependency of job B), which I believe is not supported?

**Comment by sryza** [comment, marked as answer]
> There is also the idea of using Dynamic Partitions, but I would need to independently reference partitions as dependencies for other jobs or assets (i.e. Partition.2 is a dependency of Job A, but Partition.3 is a dependency of job B), which I believe is not supported?

Hi Diego - you can accomplish this with a [SpecificPartitionsPartitionMapping](https://docs.dagster.io/_apidocs/partitions#dagster.SpecificPartitionsPartitionMapping)
